packages/grid/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts

```diff
@@ -58,6 +65,13 @@ export interface GridAggregationFunction<V = any, AV = V, FAV = AV> {
    * @default `true`
    */
   hasCellUnit?: boolean;
+  /**
+   * Function that allows to redefine the value of the cells used to generate the aggregated value.
+   * Useful for multi-column aggregation when aggregated value is generated from multiple columns.
```

Comment:
I would avoid to call this "multi-column aggregation", it took me some time to understand that it was one column with aggregated value depending on other columns not just the data of the current column.

For me "multi-column aggregation" is just having two or more items in the aggregation model (like multi filtering or multi sorting)

Reply:
what about something like this
[Screen Recording - Made with FlexClip (1).webm](https://user-images.githubusercontent.com/34584266/198849336-81741fdc-c77a-4b6c-84bf-8cccee905192.webm)
Demo: https://codesandbox.io/s/wonderful-david-px9cjr?file=/demo.tsx

Reply:
what about something like this?

[Screen Recording - Made with FlexClip (1).webm](https://user-images.githubusercontent.com/34584266/198849458-b77f560a-487f-4afb-a1fb-965d720b8eb5.webm)

Demo : https://codesandbox.io/s/wonderful-david-px9cjr?file=/demo.tsx

Reply:
@flaviendelangle 
> For me "multi-column aggregation" is just having two or more items in the aggregation model (like multi filtering or multi sorting)

Yeah, I see the confusion here. Especially because in the demo I've added in this PR there are multiple columns aggregated as well 🙃
Maybe title it as "Aggregating value from multiple fields of the rows"? What do you think?

> I think it's worth taking a look at what AG-Grid is doing
> https://www.ag-grid.com/javascript-data-grid/aggregation-custom-functions/#custom-full-row-aggregation

Hmm, I don't think I understand the relation to this PR.
What I do in this PR is covered in AG Grid here: https://www.ag-grid.com/javascript-data-grid/aggregation-custom-functions/#multi-column-aggregation

Reply:
> what about something like this?
> 
>  Screen.Recording.-.Made.with.FlexClip.1.webm 
> Demo : [codesandbox.io/s/wonderful-david-px9cjr?file=/demo.tsx](https://codesandbox.io/s/wonderful-david-px9cjr?file=/demo.tsx)

Hey @yaredtsy 
Could you explain what problem you are addressing in this demo?
This doesn't seem to be related to aggregation, but maybe I'm missing something?

Reply:
> Maybe title it as "Aggregating value from multiple fields of the rows"? What do you think?

Fine for me !

Reply:
Hey @cherniavskii 
I was looking at the AGgrid implementation and they support string formulas for `valueGetter` like 'a+b'. and I thought this would be a nice feature if it was implemented. It can also be extended for users to do the computation to add or update columns from the UI like excel. i might be off topic

Reply:
> I was looking at the AGgrid implementation and they support string formulas for valueGetter like 'a+b'

Yes, but it doesn't seem to be related to aggregation though.
We can discuss this in a separate issue if the community requests this.